cloudinit/config/schemas/schema-cloud-config-v1.json

```diff
@@ -230,6 +230,92 @@
       },
       "additionalProperties": true
     },
+    "cc_ansible": {
+      "type": "object",
+      "properties": {
+        "ansible": {
+          "type": "object",
+          "additionalProperties": false,
+          "properties": {
+            "install": {
+              "type": "boolean",
+              "default": false
+            },
+            "pull": {
+              "required": ["url", "playbook-name"],
+              "type": "object",
+              "additionalProperties": false,
+              "properties": {
+                "accept-host-key": {
+                  "type": "boolean",
+                  "default": false
+                },
+                "clean": {
+                  "type": "boolean",
+                  "default": false
+                },
+                "full": {
+                  "type": "boolean",
+                  "default": false
+                },
+                "diff": {
```

Comment:
Good catch, thanks!

I'm more in favor of exiting from the module if we can't do what the user asked rather than trying to guess their intentions. `--check --diff` in an automated call[1] might be used for checking for unexpected changes in important files (i.e. `/etc/{group,shadow,passwd}`). A false negative caused by dropping `--diff` in that scenario would be bad.

[1] particularly if they set frequency to PER_ALWAYS or if we add a long poll option

Thoughts?

Reply:
+1 on staying on the safe side